src/ctl.c

```diff
@@ -2983,6 +2985,61 @@ arena_i_retain_grow_limit_ctl(tsd_t *tsd, const size_t *mib,
 	return ret;
 }
 
+/*
+ * This function cannot be used to read and write at the same time.
+ * When writing, oldp and oldlenp must all be NULL. newp should point to a
+ * char array storing the name to be set. A name longer than ARENA_NAME_LEN
+ * will be arbitrarily cut. When reading, newp must be NULL and newlen must
+ * be 0. oldp should point to a char array whose length is no shorter than
+ * ARENA_NAME_LEN or the length of the name when it was set.
+ */
+static int
+arena_i_name_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
+    void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+	unsigned arena_ind;
+	char *name = NULL;
+
+	malloc_mutex_lock(tsd_tsdn(tsd), &ctl_mtx);
+	READ_XOR_WRITE();
```

Comment:
Remind me why this has to be XOR again?

Reply:
Originally we set this because of the life cycle of the name pointer. Yet now that we are copying the name instead of directly returning the pointer, there is no life cycle problem so I think we can allow read and write at the same time now. Should we do that?

Reply:
modified so that it can read and write at the same time since there is no concern regarding the life cycle right now.